lib/gems/pending/util/postgres_admin.rb

```diff
@@ -136,12 +136,25 @@ def self.backup_pg_compress(opts)
     opts.delete(:dbname)
 
     path = Pathname.new(opts.delete(:local_file))
-
     FileUtils.mkdir_p(path.dirname)
-    Dir.mktmpdir("vmdb_backup", path.dirname) do |dir|
-      runcmd("pg_basebackup", opts, :z => nil, :format => "t", :xlog_method => "fetch", :pgdata => dir)
-      FileUtils.mv(File.join(dir, "base.tar.gz"), path)
-    end
+
+    # Build commandline from AwesomeSpawn
+    args = {:z => nil, :format => "t", :xlog_method => "fetch", :pgdata => "-"}
+    cmd  = AwesomeSpawn.build_command_line("pg_basebackup", combine_command_args(opts, args))
+
+    # Run command in a separate thread
+    read, write   = IO.pipe
+    error         = Tempfile.new.tap(&:close).path
```

Comment:
Are you sure this will work?

Ruby will GC a tempfile if it goes out of scope so you need to keep a handle to the file until you're actually done with it or get a path without creating the file. I would be afraid of a race condition between you reading the file and ruby GCing (and removing whatever is at that path).